### ISSUE TITLE
Example failed listing bucket with objects in glacier

### ISSUE THREAD

**Issue by dustin** [brendanhay/amazonka#511]
```
[SerializeError] {
  service = S3
  status  = 200 OK
  message = Failed reading: Failure parsing ObjectVersionStorageClass from value: 'glacier'. Accepted values: standard
```

**Comment by copumpkin** [comment]
This seems like it would be addressed by #515, or possibly more broadly relaxing some of the fairly hardcore response validation approaches amazonka takes if the library continues to lag upstream changes significantly.

**Comment by endgame** [comment]
#597 relaxes the constraints somewhat by representing sum types as newtypes + pattern synonyms, and #609 is to check that all S3 storage classes are present before 2.0 RC is prepared.
